otherlibs/dune-rpc/dune_rpc.ml

```diff
@@ -45,6 +47,15 @@ module V1 = struct
 
       val disconnected : t -> unit fiber
 
+      module Subscription : sig
+        type t
+
+        val cancel : t -> unit fiber
+      end
+
+      val subscribe :
+        ?id:Id.t -> t -> 'a Sub.t -> (Subscription.t * 'a stream) fiber
```

Comment:
What about when the stream is terminated with an error? (e.g. connection drops or the RPC message is not understood)
I'm worried we'll be repeating a costly `Pipe`/`Pipe_rpc` mistake of making connection errors hard or impossible to propagate correctly with this API.

Reply:
Thanks for pointing that out. In the old interface, I intended to add callback for errors. With streams, it's less clear on how an error should be propagated. I suppose I can still add the callback?

Reply:
The issue keeps coming up in the context of Pipe_rpc, and the best solution so far is to make the type of streams awayre of the possibility of error, so `('a, 'e) stream` is a stream of values `'a` that can terminate with error `'e`.

I don't know how it'd look like in the context of this PR, but I think that's something to consider.